src/bin/julialauncher.rs

```diff
@@ -255,57 +261,65 @@ fn run_app() -> Result<i32> {
     ctrlc::set_handler(|| ())
         .with_context(|| "Failed to set the Ctrl-C handler.")?;
 
-    let mut child_process = std::process::Command::new(julia_path)
-        .args(&new_args)
-        .spawn()
-        .with_context(|| "The Julia launcher failed to start Julia.")?; // TODO Maybe include the command we actually tried to start?
-
-    run_versiondb_update(&config_file)
-        .with_context(|| "Failed to run version db update")?;
-
-    run_selfupdate(&config_file)
-        .with_context(|| "Failed to run selfupdate.")?;
-
-    let status = child_process.wait()
-        .with_context(|| "Failed to wait for Julia process to finish.")?;
-
-    let code = match status.code() {
-        Some(code) => code,
-        None => {
-            #[cfg(not(windows))]
-            {
-                use std::os::unix::process::ExitStatusExt;
-
-                let signal = status.signal();
-
-                if let Some(signal) = signal {
-                    let signal = nix::sys::signal::Signal::try_from(signal)
-                        .with_context(|| format!("Unknown signal value {}.", signal))?;
-
-                    nix::sys::signal::raise(signal)
-                        .with_context(|| "Failed to raise signal.")?;
-
-                    // We raise the signal twice because SIGSEGV and SIGBUS require that
-                    // https://github.com/JuliaLang/juliaup/pull/525#issuecomment-1353526900
-                    // https://github.com/rust-lang/rust/blob/984eab57f708e62c09b3d708033fe620130b5f39/library/std/src/sys/unix/stack_overflow.rs#L60-L80
-                    nix::sys::signal::raise(signal)
-                        .with_context(|| "Failed to raise signal.")?;
-
-                    anyhow::bail!("Maybe we should never reach this?");
-                }
-                else {
-                    anyhow::bail!("We weren't able to extract a signal, this should never happen.");
-                }
-            }
-
-            #[cfg(windows)]
-            {
-                anyhow::bail!("There is no exit code, that should not be possible on Windows.");
-            }            
+    // On *nix platforms we replace the current process with the Julia one.
+    // This simplifies use in e.g. debuggers, but requires that we fork off
+    // a subprocess to do the selfupdate and versiondb update.
+    #[cfg(not(windows))]
+    match unsafe{fork()} {
+        Ok(ForkResult::Parent { child: _, .. }) => {
+            // XXX: because we never wait on our child, it'll appear as a zombie
```

Comment:
Maybe do a double fork and the intervening child exits? I believe that's standard when starting a daemon process. See https://stackoverflow.com/questions/881388/what-is-the-reason-for-performing-a-double-fork-when-creating-a-daemon. Also allows detaching from the TTY which may or may not also be a good idea.

Reply:
Nice, that does the job! Thank for the reference, I didn't know this technique.